src/marshmallow/validate.py

```diff
@@ -105,38 +106,54 @@ class RegexMemoizer:
         def __init__(self):
             self._memoized = {}
 
-        def _regex_generator(self, relative: bool, require_tld: bool) -> typing.Pattern:
-            return re.compile(
-                r"".join(
+        def _regex_generator(
+            self, relative: bool, absolute: bool, require_tld: bool
+        ) -> typing.Pattern:
+            _absolute_part = "".join(
+                (
+                    # scheme
+                    r"(?:[a-z0-9\.\-\+]*)://",  # scheme is validated separately
+                    # basic_auth
+                    r"(?:[^:@]+?(:[^:@]*?)?@|)",  # basic auth
```

Comment:
Tanks for adding comments. Maybe put all comments in their own line for consistency.

Reply:
Sure, I'll do a quick cleanup pass.